jujugui/static/gui/src/app/components/inspector/unit-list/test-unit-list.js

```diff
@@ -441,6 +441,49 @@ describe('UnitList', () => {
     var envResolved = sinon.stub();
     var units = [{
       displayName: 'mysql/0',
+      id: 'mysql/0',
+      agent_status: 'running'
+    }, {
+      displayName: 'mysql/1',
+      id: 'mysql/1',
+      agent_status: 'running'
+    }, {
+      displayName: 'mysql/2',
+      id: 'mysql/2',
+      agent_status: 'running'
+    }];
+    // Have to use renderIntoDocument here as shallowRenderer does not support
+    // refs.
+    var output = testUtils.renderIntoDocument(
+        <juju.components.UnitList
+          acl={acl}
+          destroyUnits={destroyUnits}
+          changeState={changeState}
+          envResolved={envResolved}
+          service={service}
+          units={units}
+          whenChanged={sinon.stub()} />);
+    var checkboxes = testUtils.scryRenderedDOMComponentsWithTag(
+      output, 'input');
+    checkboxes[1].checked = true;
+    testUtils.Simulate.change(checkboxes[1]);
+    checkboxes[3].checked = true;
+    testUtils.Simulate.change(checkboxes[3]);
+    var button = testUtils.findRenderedDOMComponentWithClass(
+        output, 'button--neutral');
+    testUtils.Simulate.click(button);
+    assert.equal(destroyUnits.callCount, 1);
+    assert.deepEqual(destroyUnits.args[0][0], [units[0].id, units[2].id]);
+    // Make sure we mark the unit as resolved so that we can remove it.
+    assert.equal(envResolved.callCount, 2);
+  });
+
+  it('won\'t remove pending units', function() {
```

Comment:
We definitely still want pending units to be removable. I think it's just the test name "does not create network request to remove uncommitted units"